api/src/main/java/me/lokka30/treasury/api/common/event/Completion.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * This file is/was part of Treasury. To read more information about Treasury such as its licensing, see <https://github.com/lokka30/Treasury>.
+ */
+
+package me.lokka30.treasury.api.common.event;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.CountDownLatch;
+import java.util.function.Consumer;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public final class Completion {
+
+    public static Completion completed() {
+        return new Completion(true);
+    }
+
+    public static Completion completedExceptionally(@NotNull Throwable error) {
+        return new Completion(error);
+    }
+
+    public static Completion completedExceptionally(@NotNull List<@NotNull Throwable> errors) {
```

Comment:
Would it make more sense to accept a `Collection`? Same for the other methods that take a list